src/main/java/org/spongepowered/api/world/storage/WorldProperties.java

```diff
@@ -219,6 +220,33 @@
      */
     PortalAgentType getPortalAgentType();
 
+    /**
+     * Gets the portal agent map of this world.
+     * Keys represent dimensions names.
+     * Values represent destination world names.
+     *
+     * @return The portal agent map of this world
+     */
+    Map<String, String> getPortalAgents();
+
+    /**
+     * Set the destination world for a {@link PortalType}
+     * on this world.
+     *
+     * @param portalType The type of the portal
+     * @return The name of the destination world
+     */
+    String getPortalDestination(PortalType portalType);
```

Comment:
I think this should be an `Optional<WorldProperties>`, again mimicing what the server sees.

Reply:
Agreed.

Reply:
You talked about unknown worlds in another conversation
It's not possible to retrieve them anymore if I return Optional<WorldProperties> instead

Reply:
As API 7 basically knows about all it's loaded and unloaded worlds, that's not really an issue - it simply infers that the portal agent doesn't have a valid target in this context.

The point I was making about these methods it shouldn't mirror the config, but mirror what the server will do/knows.